### PR TITLE
Restart VPN daily

### PR DESCRIPTION
**Pull request by klinker24**
Eric let me know that he had seen a few instances of the VPN shutting down, without a restart, after days of ongoing use.

While the foreground service does give us the best chance of keeping the connection alive, Android still does not guarantee that the service can be kept alive indefinitely:

> It is still theoretically possible for the service to be killed under extreme memory pressure from the current foreground application.

---- 

This PR attempts to address this issue. Every night, at 3 AM, the service will take a few seconds to restart itself, along with the VPN connection.

Putting a WIP tag on this, so that I can watch it over the next few days to ensure that everything is working as expected, after the restarts.

### CODE REVIEW

File: app/src/main/java/com/netangel/netangelprotection/service/DailyVpnRestartService.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                VpnStateService.start(DailyVpnRestartService.this);
+                ConnectVpnActivity.start(DailyVpnRestartService.this, true);
+            }
+        }).start();
+    }
+
+    public static void scheduleNextRun(Context context) {
+        Calendar calendar = Calendar.getInstance();
+
+        // force the calendar to 3 in the morning, on the next day.
```
Comment (eterry1388) [inline app/src/main/java/com/netangel/netangelprotection/service/DailyVpnRestartService.java]: Will this take into account the timezone?

Comment (klinker24) [inline app/src/main/java/com/netangel/netangelprotection/service/DailyVpnRestartService.java]: Yes, will be fine. This only runs locally, so it just pulls in the devices local time and uses that for the timestamp, it doesn't need to worry about timezones

**Comment by klinker24** [comment]
So after a week, I haven't seen any issues with this one, although I do run NetAngel a bit differently than the average user, since I am changing the code. Every time I reinstall the app (with new changes), it would really invalidate any results here, since the service would be restarted as well.

I haven't had any issues, so far, with the VPN going out on me, but it is just difficult for me to test, because of that. Could one of you guys pull this down and watch for any issues over the next few days, as well? If ever you wake up in the morning, and the VPN isn't connected, that is where the issue would be. Let me know if you can, and thanks for any help you can give me here!

**Comment by eterry1388** [comment]
I have this on my phone now and I'll see how it goes.  Just a question, if they toggle the filter off in the app, it will stay off right?  It won't try to turn it on at 3am?

**Comment by klinker24** [comment]
@eterry1388 yep, should be fine on that regard, in [onHandleIntent](https://github.com/net-angel/netangel-protection-android/pull/49/files#diff-5ab37adce51aa4626854dcd9edc56c96R25) we check to make sure the VPN is enabled, before shutting down/restarting anything

**Pull request merged**